### PR TITLE
Very long path

### PR DESCRIPTION
**Pull request by schinagl**
### Problems
Ever since Winfile supports path lengths up to 256 characters. In certain situations it is useful to have more. Windows supports 32767, but this PR aims less and increases up to 1024
It also addresses the request from #50

### Changes
* MAX_PATH occurrences were replaced with MAXPATHLEN so that there is one common definition for the length
* Added the long longPathAware manifest
* Added verylongpath.reg file which lifts the 256 character limitation for Windows 10 system > 1607. Can be [found here] (https://github.com/microsoft/winfile/files/9175242/TestVeryLongPath.zip)

* Increased the stacksize so that filling the Goto-Cache survives 512 levels of recursion
* Fixed a bug which was there anyhow via  [44991cb]

### How to test
Take [TestDeepPath.bat](https://github.com/microsoft/winfile/files/9175242/TestVeryLongPath.zip) and run it. A path with about 750 chars is created
Now play around, and navigate through it. Do all you think should work

### Status
Basically this PR works quite well, but it is very experimental, because a change like this 
* is hard to test, because all tests are manual
* the achieved test coverage is unknown.
So it is a draft PR, just for all who want to compile their own Winfile and maybe it works

### CODE REVIEW

**Comment by schinagl** [comment]
Had Winfile.exe with 1024 characters running for a week during my everyday use, and it worked flawlessly. 
There seem to be no sideeffects on the operations <= 256 characters and the operations > 256 also work really well. Who ever wrote this thing 30 years ago and implemented the limits in WinFile, really did it very well :-)

Anyhow it would be nice if you could help testing this. A version from my [private](https://github.com/schinagl/winfile/tree/_hermann) branch
[can be downloaded here](https://schinagl.priv.at/nt/winfile/winfile.zip). I called this version 10.1.4.42. This is just **my** internal number.

Most likley there will be an official version with precompiled binaries having many changes from the last year in during the course of this year.

Thx for your help in testing this.

**Comment by schinagl** [comment]
Rebased onto #341 merge to master

File: src/wfcomman.c [hunk 1]
```diff
@@ -601,7 +601,7 @@ OpenOrEditSelection(HWND hwndActive, BOOL fEdit)
    DWORD ret;
    HCURSOR hCursor;
 
-   WCHAR szPath[MAXPATHLEN+2];  // +2 for quotes if needed
+   WCHAR szPath[MAXPATHLEN * 2 + 2];  // +2 for quotes if needed, * 2 because CreateDirWindow() assumes a filename can be the same length as MAXPATHLEN
```
Comment (craigwims) [inline src/wfcomman.c:605]: Let's separate out this change per #344 .

**Comment by schinagl** [comment]
Resolved conflicts solution brought in with #344
Basically it now comes down to a few very simple things
- Replace MAX_PATH with MAXPATHLEN in a few places, because MAXPATHLEN is the winfile-way to express this.
- Added the longPathAware manifest to the .manifest file
- Increased the stack size in the .vcxproj so that filling the Goto-Cache survives 512 levels of recursion. (tried it)

Voila, and this works perfect since months on my site

**Comment by schinagl** [comment]
Any news here? Can we add this since it almost contains no code changes anymore.... ?

**Comment by craigwims** [comment]
reviewing again.

**Comment by craigwims** [comment]
I'm good with this PR.  Shall I merge it now or wait for #312?

**Comment by schinagl** [comment]
Merge it. Will rebase 312 if needed
